### PR TITLE
Jdb/sort-chart-series

### PR DESCRIPTION
**Pull request by unclejustin**
This PR sorts the results for insight series data by semantic version first then alphabetically.

Closes https://github.com/sourcegraph/sourcegraph/issues/29949

## Before

PixelSnap 2022-04-05 at 16 19 49

## After 
PixelSnap 2022-04-05 at 16 22 25

## Test plan

Unit tests have been added to test the sorting function.

The updated results of this can be seen in storybook under `/?path=/story/web-insights-smartinsightsviewgridexample--smart-insights-view-grid-example` and in the screenshots above.

To manually test you need to create an insight with many results. A "Detect and Track" insight would be the best choice.

## App preview:
- [Link](https://sg-web-jdb-sort-chart-series.onrender.com)

### CODE REVIEW

File: client/web/src/enterprise/insights/core/backend/utils/create-line-chart-content.ts [hunk 1]
```diff
@@ -40,37 +42,45 @@ export function createLineChartContent(
...
+                            const beforeFilter = `before:${before}`
+                            const dateFilters = `${afterFilter} ${beforeFilter}`
+                            const diffQuery = `${repoFilter} type:diff ${dateFilters} ${
+                                definitionMap[line.seriesId].query
+                            }`
+                            const searchQueryParameter = buildSearchURLQuery(
+                                diffQuery,
+                                SearchPatternType.literal,
+                                false
+                            )
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/core/backend/utils/create-line-chart-content.ts:77]: All of this is just indented.

File: client/web/src/insights/utils/semantic-sort.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { semanticSort } from './semantic-sort'
+
+describe('sortSeriesByName', () => {
```
Comment (vovakulikov) [inline client/web/src/insights/utils/semantic-sort.test.ts:3]: 👍 for units

File: client/web/src/insights/utils/semantic-sort.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import semver from 'semver'
```
Comment (vovakulikov) [inline client/web/src/insights/utils/semantic-sort.ts:1]: I saw that we've been using this package on Admin Page for parsing Sourcegraph version as well, but I was amazed that this package is around 11kb in gzip and this means this is 33kb in runtime. Just for the parsing version (I'm sure this isn't so easy but 33kb looks like overkill) Can we create a follow-up for this and replace it in the future?

**Review by vovakulikov — APPROVED** [review]
Left a few comments, nothing big. Thanks @unclejustin

File: client/web/src/insights/utils/semantic-sort.ts (hunk 3 above)
Comment (unclejustin) [reply client/web/src/insights/utils/semantic-sort.ts:1]: Wow! Great catch! Yet another argument to handle this sorting on the backend.

@sourcegraph/code-insights-backend is it feasible to sort data series on the backend instead? 33kb is a pretty big chunk to download just for this.

**Pull request merged**